build.gradle.kts

```diff
@@ -10,6 +11,15 @@ subprojects {
                 jvmTarget = "17"
             }
         }
+        withType<ProcessResources> {
```

Comment:
I wasn't able to get the version in the same was that the java client does.  It gets the version at runtime using ` IceAdapter.class.getPackage().getImplementationVersion();`  In kotlin-ice-adapter `getPackage()` always returns null.  Searching around for how to get the `version` from gradle I saw several solutions like [this one](https://stackoverflow.com/questions/50117966/get-version-in-kotlin-java-code-in-gradle-project) which adds a properties file to build/resources which contains the version. I think this is kind of ugly but I wasn't able to get `getPackage().getImplementationVersion()` working.

Reply:
That is a known issue and solved e.g. in the FAF client.

To solve this you need to cheat a little:
1. [Declare the version in build.gradle referencing a variable](https://github.com/FAForever/downlords-faf-client/blob/dd9ab2b128a4da6a90fa885a05de4022b43d83f7/build.gradle#L24)
2. [Set a default version in the properties](https://github.com/FAForever/downlords-faf-client/blob/dd9ab2b128a4da6a90fa885a05de4022b43d83f7/gradle.properties#L1)
3. [Set the version in the CI pipeline](https://github.com/FAForever/downlords-faf-client/blob/dd9ab2b128a4da6a90fa885a05de4022b43d83f7/.github/workflows/create-release.yml#L100) (i.e. derive from release tag)

I have no working example in gradle kotlin style though.

Reply:
Ahh got it.  I added a commit which sets the default version in gradle.properties.  Looking at the faf-client I see that the version is also added to the manifest so I also added that to the `shadowJar` task.  I think that's the main piece I was missing. With that in place, when I run the jar it can get the version that is set in gradle.properties. 

I didn't do the CI pipeline part. This makes sense to me generally, but I'm not at all familiar with how this release is made (or should be made?) It doesn't look like there is currently any release CI defined that I can just add the versioning bit to.  Adding the release yml seems out of scope for this PR at least. I can look into doing that if you want, but I've never messed with that kind of thing before.

Reply:
Also, I guess the production java-ice-adapter doesn't seem to get the version correctly. The telemetry ui just shows `java-ice-adapter/SNAPSHOT` which is the default string if it isn't getting the version.  🤷